src/coreclr/jit/codegenxarch.cpp

```diff
@@ -491,12 +491,23 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
 
             if (vecCon->IsAllBitsSet())
             {
-                if ((attr != EA_32BYTE) || compiler->compOpportunisticallyDependsOn(InstructionSet_AVX2))
+                if ((attr == EA_16BYTE) ||
+                    ((attr == EA_32BYTE) && compiler->compOpportunisticallyDependsOn(InstructionSet_AVX2)))
                 {
 #if defined(FEATURE_SIMD)
                     emit->emitIns_SIMD_R_R_R(INS_pcmpeqd, attr, targetReg, targetReg, targetReg);
 #else
                     emit->emitIns_R_R(INS_pcmpeqd, attr, targetReg, targetReg);
+#endif // FEATURE_SIMD
+                    break;
+                }
+                else if (attr == EA_64BYTE)
+                {
+                    assert(compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F));
+#if defined(FEATURE_SIMD)
+                    emit->emitIns_SIMD_R_R_R_I(INS_vpternlogd, attr, targetReg, targetReg, targetReg, (int8_t)0xFF);
```

Comment:
nit: We've been trying to standardize on `static_cast<int8_t>(0xFF)`.

I don't recall the exact reasoning, just that its been a frequent review request anytime I forget to do it myself 😅